# **Data Structures**

## **Question 1. List and Tuple**

**Create a list named fruits containing the following: 'apple', 'banana', 'cherry'.**

In [14]:
fruits={'apple','banana','cherry'}

**Convert this list into a tuple.**

In [15]:
fruits_tuple=tuple(fruits)

**Add 'orange' to the list of fruits.**

In [16]:
fruits.add('orange')

**Show the difference in the content of the list and tuple after addition.**

In [17]:
d=fruits.difference(fruits_tuple)
print(d)

{'orange'}


## **Question 2. Dictionary**

**Create a dictionary named person with keys 'first_name', 'last_name', and 'age'.**

**Populate it with values of your choice.**

In [ ]:
person={'first_name':'Hala','last_name':'Muhiar','age':20}

**Add a new key-value pair 'city' with a value of your choice.**

In [ ]:
person['city']='Dallas'
person

{'first_name': 'Hala', 'last_name': 'Muhiar', 'age': 20, 'city': 'Dallas'}

**Display the keys in the dictionary using a Python command.**

In [ ]:
person.keys()

dict_keys(['first_name', 'last_name', 'age', 'city'])

## **Question 3. Set**

**Create a set named colors with the values 'red', 'blue', 'green'.**

In [ ]:
colors=set(['red','blue','green'])

**Add 'yellow' to the set and remove 'blue'.**

In [ ]:
colors.add('yellow')
colors.remove('blue')
print(colors)

{'red', 'green', 'yellow'}


**Check if 'pink' is in the set and display the result.**

In [ ]:
'pink' in colors

False

---

# **Conditional Statements**

## **Question 4**

**Write a Python program that takes an input from the user and checks if the number is positive, negative, or zero.**

In [ ]:
n=int(input('Enter a number: '))
if n>0:
  print('Your number is positive.')
elif n<0:
  print('Your number is negative: ')
else:
  print('Your number is zero.')

Enter a number: 3
Your number is positive.


## **Question 5**

**Given a list of numbers: [23, 45, 8, 12, 78, 56, 89], write a Python program that prints out all even numbers from the list using a conditional statement.**

In [ ]:
numbers=[23,45,8,12,78,56,89]

In [ ]:
for n in numbers:
  if n%2==0:
    print(n)

8
12
78
56


## **Question 6**

**Write a program that asks the user to enter their age. Based on the user's input, determine if they are eligible to vote.**

In [ ]:
vote_age=int(input('Enter your age: '))
if vote_age>=18:
  print('You are eligible to vote.')
else:
  print('You are not eligible to vote.')

Enter your age: 20
You are eligible to vote.


---

# **While Loop**

## **Question 7**

**Write a program that prompts the user to enter numbers continuously (using a while loop) and stops taking inputs when the user enters 'q'.**

**After that, display the sum of the numbers entered.**

In [23]:
Sum=0

while True:
  num=input('Enter a number (or q to quit): ')
  if num=='q':
    break
  else:
    num1=int(num)
    Sum+=num1
print(Sum)

Enter a number (or q to quit): 1
Enter a number (or q to quit): 2
Enter a number (or q to quit): 3
Enter a number (or q to quit): 4
Enter a number (or q to quit): 5
Enter a number (or q to quit): q
15


## **Question 8**

**Create a Python program using a while loop that prints numbers from 10 down to 1 and then prints 'Liftoff!'**

In [ ]:
n=10

while n>0:
  print(n)
  n=n-1
print('Liftoff!')

10
9
8
7
6
5
4
3
2
1
Liftoff!


## **Question 9**

**Create a program that asks the user for a number and then prints out the multiplication table for that number using a while loop.**

In [ ]:
k=int(input('Enter a number: '))
j=0

print('Multiplication Table for',k)
while j<=12:
  j=j*1
  print(k,'x',j,'=',(k*j))
  j+=1

Enter a number: 12
Multiplication Table for 12
12 x 0 = 0
12 x 1 = 12
12 x 2 = 24
12 x 3 = 36
12 x 4 = 48
12 x 5 = 60
12 x 6 = 72
12 x 7 = 84
12 x 8 = 96
12 x 9 = 108
12 x 10 = 120
12 x 11 = 132
12 x 12 = 144


---

# **For Loop**

## **Question 10**

**Given a list: ['apple', 'banana', 'cherry', 'date'], use a for loop to print each fruit in the list.**

In [ ]:
fruits2=['apple','banana','cherry','date']

for f in fruits2:
  print(f)

apple
banana
cherry
date


## **Question 11**

**Using a for loop, calculate the factorial of a given number (n!). For example, if the given number is 5, the output should be 5 x 4 x 3 x 2 x 1 = 120.**

In [ ]:
user_num=int(input('Enter a number: '))
fact=1

if user_num==0:
  print('The factorial of zero is one.')
elif user_num<0:
  print('Cannot compute.')
else:
  for g in range(1,user_num+1):
    fact=fact*g
  print(f'{user_num}!=',fact)

Enter a number: 7
7!= 5040


## **Question 12**

**Create a program using a for loop that prompts the user to enter five numbers. After all the numbers are entered, the program should display the average of the numbers.**

In [21]:
q=int(input('Enter a number: '))
w=int(input('Enter a number: '))
e=int(input('Enter a number: '))
r=int(input('Enter a number: '))
t=int(input('Enter a number: '))

y=[q,w,e,r,t]
o=len(y)
a=q+w+e+r+t

for h in y:
  w=a/o
print(a/o)

Enter a number: 2
Enter a number: 4
Enter a number: 6
Enter a number: 8
Enter a number: 10
6.0
